packages/fixie/src/agent.ts

```diff
@@ -475,8 +481,28 @@ export class FixieAgent {
       throw Error(`No package.json found in ${packageJsonPath}. Only JS-based agents are supported.`);
     }
 
+    const packageJson = fs.readFileSync(packageJsonPath, 'utf8');
+    let json;
+    try {
+      json = JSON.parse(packageJson);
+    } catch (ex) {
+      throw Error(`Error parsing package.json: ${ex}`);
+    }
+
+    if (json.scripts) {
+      const scriptName = json.scripts.prepack ? 'prepack' : json.scripts.build ? 'build' : undefined;
```

Comment:
To be fully pedantic, I think you may need to check for a few more keys: https://docs.npmjs.com/cli/v10/using-npm/scripts#npm-pack.

Ultimately, the safest thing to do is just to run `npm publish` and let npm invoke the right thing. But I recognize that makes it harder to do what we're trying to do here of presenting a nicer DX, so maybe we can't do that.

Reply:
I buy it -- and it's also simpler to just run the same `npm pack` function we use for `fixie deploy`.